### PR TITLE
Switch to go modules

### LINKED ISSUE #43: migrate to go modules

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Signed-off-by: Ashish Amarnath ashisham@vmware.com

Closes #43

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,10 +13,9 @@
 # limitations under the License.
 
 FROM golang:1.13-buster AS build
-WORKDIR /go/src/github.com/vmware-tanzu/velero-plugin-for-aws
-# copy vendor in separately so the layer can be cached if the contents don't change
-COPY vendor vendor
-COPY velero-plugin-for-aws velero-plugin-for-aws
+COPY . /go/src/velero-plugin-for-aws
+WORKDIR /go/src/velero-plugin-for-aws
+RUN go mod download
```
Comment (skriss) [inline Dockerfile:18]: is there a specific reason to include this step?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	github.com/hashicorp/go-hclog v0.9.2 // indirect
+	github.com/hashicorp/go-plugin v1.0.1-0.20190610192547-a1bc61569a26 // indirect
+	github.com/hashicorp/yamux v0.0.0-20190923154419-df201c70410d // indirect
+	github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
+	github.com/mitchellh/go-testing-interface v1.0.0 // indirect
+	github.com/pkg/errors v0.9.1
+	github.com/sirupsen/logrus v1.4.2
+	github.com/spf13/pflag v1.0.5
+	github.com/stretchr/testify v1.5.1
+	github.com/vmware-tanzu/velero v1.3.1-0.20200501201116-577af5a5b1a6
```
Comment (skriss) [inline go.mod:17]: we'll want to update this to v1.4.0 now

File: hack/build.sh [hunk 3]
```diff
@@ -45,6 +45,8 @@ if [[ "${GOOS}" = "windows" ]]; then
   OUTPUT="${OUTPUT}.exe"
 fi
 
+go mod download
```
Comment (skriss) [inline hack/build.sh:48]: is there a specific reason to call this here?

File: Dockerfile (hunk 1 above)
Comment (ashish-amarnath) [reply Dockerfile]: No, removed.

File: hack/build.sh (hunk 3 above)
Comment (ashish-amarnath) [reply hack/build.sh]: done!

File: go.mod (hunk 2 above)
Comment (ashish-amarnath) [reply go.mod]: done

**Pull request merged**